finder-app/writer.c

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************************
+   File Name           : writer.c
+   Author Name         : Kshitija Dhondage 
+   Compiler            : gcc and aarch64-none-linux-gnu-gcc 
+   Date                : 29th January 2023
+   references:         :https://stackoverflow.com/questions/23092040/how-to-open-a-file-which-overwrite-existing-content/23092113
+  
+*******************************************************************************************/
+
+/*includes*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <syslog.h>
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+/*Macros*/
+#define TOTAL_ARGC         ( 3  )
+#define FILE_PERMMISSION   (0644)
+#define FAIL               ( -1 )
+
+int main(int argc, char *argv[])
+{
+
+	char *writeFile;
+	char *writeStr;
+	
+	int fd;
+	int ret_status = 0;
+	
+	/* Initialize syslog connection */
+	openlog(NULL, 0, LOG_USER);
+
+        /* Validate arguments */
+	if (argc != TOTAL_ARGC)
+	{
+		syslog(LOG_ERR, "Invalid arguments\n");
+		syslog(LOG_INFO, "Argument 1 < File path>\n");
+		syslog(LOG_INFO, "Argument 2 String to be added \n");
+		exit(1);
+	}
+
+       /* Copy arguments in local variables */
+	writeFile = argv[1];
+	writeStr  = argv[2];
```

Comment:
I like the idea to write to a variable.  Check out the class discussion about const correct from today.